In [4]:
import os, json, sys
sys.path.append(os.path.abspath(os.path.join('..')))
import pandas as pd

from seismic_graph import Study
from seismic_graph.format_conversion.seismic_format import SEISMICdata

import logging
logging.basicConfig(level=logging.DEBUG, format='%(asctime)s:%(levelname)s:%(name)s:%(message)s')
logger = logging.getLogger()

In [5]:
data = []
# paths = [
# '/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/out/LIB063158_TRA00287226_2024-08-09-ML2-1-1_S1_L001__webapp.json',
# '/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/out/LIB063158_TRA00287226_2024-08-09-ML2-1-1_S1_L002__webapp.json',
# '/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/out/LIB063158_TRA00287227_2024-08-09-ML2-1-2_S2_L001__webapp.json',
# '/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/out/LIB063158_TRA00287227_2024-08-09-ML2-1-2_S2_L002__webapp.json',
# '/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/out/LIB063158_TRA00287228_2024-08-09-ML2-1-0_S3_L001__webapp.json',
# '/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/out/LIB063158_TRA00287228_2024-08-09-ML2-1-0_S3_L002__webapp.json',
# '/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/out/LIB063158_TRA00287229_2024-08-09-ML2-2-1_S4_L001__webapp.json',
# '/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/out/LIB063158_TRA00287230_2024-08-09-ML2-2-2_S5_L001__webapp.json',
# '/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/out/LIB063158_TRA00287230_2024-08-09-ML2-2-2_S5_L002__webapp.json',
# '/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/out/LIB063158_TRA00287231_2024-08-09-ML2-2-0_S6_L001__webapp.json',
# '/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/out/LIB063158_TRA00287231_2024-08-09-ML2-2-0_S6_L002__webapp.json'
# ]

# paths = [
#     '/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/modified copy/long_control_L001.json',
#     '/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/modified copy/short_control_L002.json',
#     '/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/modified copy/long_control_L002.json',
#     '/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/modified copy/short_dms_rep1_L001.json',
#     '/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/modified copy/long_dms_rep1_L001.json',
#     '/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/modified copy/short_dms_rep1_L002.json',
#     '/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/modified copy/long_dms_rep2_L001.json',
#     '/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/modified copy/short_dms_rep2_L001.json',
#     '/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/modified copy/long_dms_rep2_L002.json',
#     '/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/modified copy/short_dms_rep2_L002.json',
#     '/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/modified copy/short_control_L001.json'
# ]

paths = [
    '/Users/casper/Documents/data/Adam/Astrophage/AstrophagePooled/AstrophagePooled__webapp.json'
]


for path in paths:
    with open (path, 'r') as file:
        json_data = json.load(file) 
        data.append(json_data)

study = Study(data=data)
sample, reference, section = study.df.iloc[0][['sample', 'reference', 'section']]
df = study.df
column_headers = study.df.columns.tolist()
print(column_headers)

['sample', 'reference', 'num_aligned', 'sequence', 'section', 'cluster', 'section_end', 'section_start', 'positions', 'deltaG', 'structure', 'cov', 'del', 'info', 'ins', 'sub_A', 'sub_C', 'sub_G', 'sub_N', 'sub_T', 'sub_hist', 'sub_rate', 'min_cov']


In [4]:
df['num_aligned'].sum()

42425162

Astrophage Pooled total aligned reads: 42,425,162

Astrophage unpooled total aligned reads: 39,971,301

In [10]:
filtered_df = df[df['min_cov'] == 0]
filtered_df

sample  \
13    LIB063158_TRA00287226_2024-08-09-ML2-1-1_S1_L001   
22    LIB063158_TRA00287226_2024-08-09-ML2-1-1_S1_L001   
27    LIB063158_TRA00287226_2024-08-09-ML2-1-1_S1_L001   
32    LIB063158_TRA00287226_2024-08-09-ML2-1-1_S1_L001   
34    LIB063158_TRA00287226_2024-08-09-ML2-1-1_S1_L001   
...                                                ...   
3655  LIB063158_TRA00287231_2024-08-09-ML2-2-0_S6_L002   
3659  LIB063158_TRA00287231_2024-08-09-ML2-2-0_S6_L002   
3665  LIB063158_TRA00287231_2024-08-09-ML2-2-0_S6_L002   
3672  LIB063158_TRA00287231_2024-08-09-ML2-2-0_S6_L002   
3679  LIB063158_TRA00287231_2024-08-09-ML2-2-0_S6_L002   

                      reference family  num_aligned  \
13                   srp_Heli_9    srp         1067   
22                     srp_Pedi    srp         1123   
27                     srp_Rick    srp         1137   
32                  srp_Stre_16    srp         1030   
34                   srp_Stre_7    srp         1288   
...                         ...    ...          ...   
3655       5s_Ustilago-maydis-1     5s         1095   
3659  5s_Zygoascus-hellenicus-1     5s         1588   
3665                   srp_Anab    srp         1199   
3672                 srp_Meth_9    srp         1071   
3679                 srp_Rhod_6    srp         1017   

                                               sequence section  cluster  \
13    AATGCGTTAGGCTGGTTTCACAGAGCTGCGAGAACCTCACGCTCTA...    full  average   
22    AGGCAAGGTTGGACTTACGAATTGTGTCAGGTCCGGAAGGAAGCAG...    full  average   
27    GCTAGTAGTGGGCATTGCTCTTGCTTAGTTGGTCAGGTCTGAAAAG...    full  average   
32    GGAGCAACAGTTTTGCGTGAAGCGGGTCAGGGGAGGAATCCAGCAG...    full  average   
34    AAAAGTGGAGCAACACTTGTGCGTGAAGTGGGTCAGGGGAGGAATC...    full  average   
...                                                 ...     ...      ...   
3655  ATCTGCGGCCACAGAGACTTGAAAATACCGCATCCCGTCCGATCTG...    full  average   
3659  AGCTACGGCCATATCCAGATGAAAATAAGACTTCCCGTCCGATCAG...    full  average   
3665  AAGCCTGGGCCCGTGCGGCTCGGACGCCCGAACCGTGTCAGGACCT...    full  average   
3672  AGGCTAGGCCGGGGGGTTAGGGGTCCCCTGTAAGCGCAAATCCCCT...    full  average   
3679  GGCCCCTACGTGGTGTCATCTCGCTGAACTTCCCCCAGGACCGGAA...    full  average   

      section_end  section_start  \
13             64              1   
22             59              1   
27             87              1   
32             73              1   
34             89              1   
...           ...            ...   
3655          118              1   
3659          120              1   
3665          101              1   
3672          154              1   
3679           97              1   

                                              positions  ...  \
13    [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14...  ...   
22    [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14...  ...   
27    [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14...  ...   
32    [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14...  ...   
34    [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14...  ...   
...                                                 ...  ...   
3655  [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14...  ...   
3659  [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14...  ...   
3665  [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14...  ...   
3672  [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14...  ...   
3679  [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14...  ...   

                                                   info  \
13    [nan, nan, nan, nan, nan, nan, nan, nan, nan, ...   
22    [nan, nan, nan, nan, nan, nan, nan, nan, nan, ...   
27    [nan, nan, nan, nan, nan, nan, nan, nan, nan, ...   
32    [nan, nan, nan, nan, nan, nan, nan, nan, nan, ...   
34    [nan, nan, nan, nan, nan, nan, nan, nan, nan, ...   
...                                                 ...   
3655  [nan, nan, nan, nan, nan, nan, nan, nan, nan, ...   
3659  [nan, nan, nan, nan, nan, nan, nan, nan, nan, ...   
3665  [nan, nan, nan, nan, nan, nan, nan, na

In [31]:
study.df['reference'].nunique()

1764

Astrophage Unpooled has 929 unique references

Astrophage Pooled has 1764 unique references


Astrophage Unpooled has 3686 rows

Astrophage Pooled has 1764 rows

In [32]:
df = study.df
len(df)

1764

In [33]:
def count_sample_references(df):
    # Group by 'sample' and aggregate
    result = df.groupby('sample').agg(
        unique_references=('reference', 'nunique'),  # Count unique references
    ).reset_index()
    
    return result

result_df = count_sample_references(df)
print(result_df)

             sample  unique_references
0  AstrophagePooled               1764


In [9]:
# Unpooled
fig = study.dist_of_seq_lengths(
    # min_cov = 1000,
    # sample=['8_1_S2_L001','8_2_S3_L001'],
    # sample=['8_2_S3_L001'],
    # reference = 'ABCA4',
    # section = ['full'],
)['fig']
fig.show()

In [34]:
# Pooled
fig = study.dist_of_seq_lengths(
    # min_cov = 1000,
    # sample=['8_1_S2_L001','8_2_S3_L001'],
    # sample=['8_2_S3_L001'],
    # reference = 'ABCA4',
    # section = ['full'],
)['fig']
fig.show()

In [39]:
# Unpooled
fig = study.percent_masked_histogram(
    # min_cov = 1000,
    to_html="/Users/casper/Documents/output/pct_AC_masked.html"
    # sample=['8_1_S2_L001','8_2_S3_L001'],
    # sample=['8_2_S3_L001'],
    # reference = 'ABCA4',
    # section = ['full'],
)['fig']
fig.show()

In [35]:
# Pooled
fig = study.percent_masked_histogram(
    # min_cov = 1000,
    to_html="/Users/casper/Documents/output/pct_AC_masked.html"
    # sample=['8_1_S2_L001','8_2_S3_L001'],
    # sample=['8_2_S3_L001'],
    # reference = 'ABCA4',
    # section = ['full'],
)['fig']
fig.show()

In [40]:
# Pooled
fig = study.num_aligned_reads_per_reference_frequency_distribution(
    # min_cov = 1000,
    to_html="/Users/casper/Documents/output/hist_aligned_reads.html"
    # sample=['8_1_S2_L001','8_2_S3_L001'],
    # sample=['8_2_S3_L001'],
    # reference = 'ABCA4',
    # section = ['full'],
)['fig']
fig.show()

In [41]:
# Unpooled
fig = study.num_aligned_reads_per_reference_frequency_distribution(
    # min_cov = 1000,
    to_html="/Users/casper/Documents/output/hist_aligned_reads.html"
    # sample=['8_1_S2_L001','8_2_S3_L001'],
    # sample=['8_2_S3_L001'],
    # reference = 'ABCA4',
    # section = ['full'],
)['fig']
fig.show()

In [4]:
paths = [
'/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/modified/LIB063158_TRA00287226_2024-08-09-ML2-1-1_S1_L001__webapp.json',
'/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/modified/LIB063158_TRA00287226_2024-08-09-ML2-1-1_S1_L002__webapp.json',
'/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/modified/LIB063158_TRA00287227_2024-08-09-ML2-1-2_S2_L001__webapp.json',
'/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/modified/LIB063158_TRA00287227_2024-08-09-ML2-1-2_S2_L002__webapp.json',
'/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/modified/LIB063158_TRA00287228_2024-08-09-ML2-1-0_S3_L001__webapp.json',
'/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/modified/LIB063158_TRA00287228_2024-08-09-ML2-1-0_S3_L002__webapp.json',
'/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/modified/LIB063158_TRA00287229_2024-08-09-ML2-2-1_S4_L001__webapp.json',
'/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/modified/LIB063158_TRA00287230_2024-08-09-ML2-2-2_S5_L001__webapp.json',
'/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/modified/LIB063158_TRA00287230_2024-08-09-ML2-2-2_S5_L002__webapp.json',
'/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/modified/LIB063158_TRA00287231_2024-08-09-ML2-2-0_S6_L001__webapp.json',
'/Users/casper/Documents/data/Adam/Astrophage/Astrophage_Data_P3/modified/LIB063158_TRA00287231_2024-08-09-ML2-2-0_S6_L002__webapp.json'
]

data = []

for path in paths:
    with open (path, 'r') as file:
        json_data = json.load(file) 
        data.append(json_data)

study = Study(data=data)
sample, reference, section = study.df.iloc[0][['sample', 'reference', 'section']]

column_headers = study.df.columns.tolist()
print(column_headers)

['sample', 'reference', 'family', 'num_aligned', 'sequence', 'section', 'cluster', 'section_end', 'section_start', 'positions', 'F1', 'deltaG', 'structure', 'cov', 'del', 'info', 'ins', 'sub_A', 'sub_C', 'sub_G', 'sub_N', 'sub_T', 'sub_hist', 'sub_rate', 'min_cov']


In [37]:
df = study.df
df['num_aligned'].dropna().max()

476762

In [8]:
fig = study.f1_violin_by_family(
    min_cov = 1000,
    # sample=['8_1_S2_L001','8_2_S3_L001'],
    # sample=['8_2_S3_L001'],
    # reference = 'ABCA4',
    # section = ['full'],
)['fig']
fig.show()

In [3]:
fig = study.percent_masked_histogram(
    to_png="/Users/casper/Documents/output/astrophage_Q20/pct_masked_hist_Q25.png",
    min_cov = 0,
)['fig']
fig.show()

{'AstrophagePooled': 0        13.636364
1        15.384615
2        28.767123
3       100.000000
4        13.333333
           ...    
1759     16.216216
1760     19.444444
1761     19.444444
1762     18.918919
1763    100.000000
Name: masked_percentage, Length: 1764, dtype: float64}
['AstrophagePooled']


In [6]:
fig = study.auroc_histogram()['fig']
fig.show()

sub_rate=array([4.91270e-02, 8.47340e-02, 1.09150e-02, 3.36700e-03, 4.65300e-03,
       3.62040e-02, 9.85200e-03, 3.75100e-02, 3.43549e-01, 1.98700e-03,
       4.77000e-04, 5.27000e-04, 9.65000e-04, 9.61000e-04, 1.95755e-01,
       1.32560e-01, 1.33500e-03, 6.83000e-04, 7.55000e-04, 5.35540e-02,
       9.23590e-02, 9.34500e-03, 9.26910e-02, 1.55080e-01, 3.39000e-04,
       1.50000e-04, 5.53000e-04, 2.56000e-04, 3.72000e-04, 6.47000e-04,
       5.03000e-04, 3.82200e-03, 2.84438e-01, 1.65660e-02, 1.36540e-02,
       3.58455e-01, 1.50000e-03, 1.01300e-03, 1.53100e-03, 6.27000e-04,
       1.84900e-03, 2.29598e-01, 5.62700e-03, 3.11400e-03, 8.15120e-02,
       9.11670e-02, 7.53580e-02, 1.11224e-01, 1.10375e-01, 1.47643e-01,
       1.35588e-01, 1.05231e-01, 8.35310e-02, 9.32300e-03, 1.01040e-02,
       6.44900e-02, 8.45690e-02])
structure_array=array([1, 1, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 1, 1, 0,
       1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0,
   